kafka-workitem/src/main/java/org/jbpm/process/workitem/kafka/KafkaWorkItemHandler.java

```diff
@@ -0,0 +1,132 @@
+import org.jbpm.process.workitem.core.util.service.WidService;
+import org.kie.api.runtime.process.WorkItem;
+import org.kie.api.runtime.process.WorkItemManager;
+import org.kie.internal.runtime.Cacheable;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+
+@Wid(widfile = "KafkaWorkItem.wid", name = "KafkaPublishMessages",
+     displayName = "KafkaPublishMessages",
+     defaultHandler = "mvel: new org.jbpm.process.workitem.kafka.KafkaWorkItemHandler()",
+     documentation = "${artifactId}/index.html",
+     parameters = {
+                   @WidParameter(name = "Topic", required = true),
+                   @WidParameter(name = "Key", required = true),
+                   @WidParameter(name = "Value", required = true)
+     },
+     results = {
+                @WidResult(name = "Result")
+     },
+     mavenDepends = {
+                     @WidMavenDepends(group = "${groupId}", artifact = "${artifactId}", version = "${version}")
+     },
+     serviceInfo = @WidService(category = "${name}", description = "${description}",
+                               keywords = "kafka,puslish,message,topic",
+                               action = @WidAction(title = "Publish message to a kafka topic")
+     ))
+
+public class KafkaWorkItemHandler extends AbstractLogOrThrowWorkItemHandler implements Cacheable {
+
+    private static final Logger LOG = LoggerFactory.getLogger(KafkaWorkItemHandler.class);
+
+    private Producer<Long, String> producer;
+
+    public KafkaWorkItemHandler() {
+
+    }
+
+    public KafkaWorkItemHandler(String bootstrapServers, String client_id, String keySerializerClass, String valueSerializerClass) {
+        Properties config = new Properties();
+        config.put(ProducerConfig.BOOTSTRAP_SERVERS_CONFIG, bootstrapServers);
+        config.put(ProducerConfig.CLIENT_ID_CONFIG, client_id);
+        config.put(ProducerConfig.KEY_SERIALIZER_CLASS_CONFIG, keySerializerClass);
+        config.put(ProducerConfig.VALUE_SERIALIZER_CLASS_CONFIG, valueSerializerClass);
+        producer = new KafkaProducer<Long, String>(config);
+    }
+
+
+    public void executeWorkItem(WorkItem workItem, WorkItemManager manager) {
+
+        try {
+            RequiredParameterValidator.validate(this.getClass(),
+                                                workItem);
+
+            Map<String, Object> results = new HashMap<String, Object>();
+            String topic = (String) workItem.getParameter("Topic");
+            String key = (String) workItem.getParameter("Key");
+            String value = (String) workItem.getParameter("Value");
+
+            try {
+
+                producer.send(new ProducerRecord(topic, key, value));
+                results.put("Result", "success");
+                manager.completeWorkItem(workItem.getId(), results);
+            } catch (Exception e) {
+                LOG.error("Kafka error", e);
+                producer.flush();
+                producer.close();
```

Comment:
Do you really want to close producer here? Maybe flush will be enough as closing it means will cause all subsequent request fail